sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -132,23 +132,19 @@ case class BroadcastHashJoinExec(
         } else if (hashed == HashedRelationWithAllNullKeys) {
           Iterator.empty
         } else {
-          val keyGenerator = UnsafeProjection.create(
-            BindReferences.bindReferences[Expression](
-              leftKeys,
-              AttributeSeq(left.output))
-          )
-          streamedIter.filter(row => {
+          val keyGenerator = streamSideKeyGenerator()
```

Comment:
It's actually same as `streamSideKeyGenerator()` since NAAJ requires `BuildRight`, then `streamedKeys = leftKeys`

Reply:
+1, it acutally prevents the potential issue that all keys in shuffle join should rewrite as LongType if possible. Previous, the hash relation has rewritten but the probe key did not. It's not a bug as NAAJ now only supports one column.